Fixed default value leng on text fieldtype and nullable flag on shape…

sainsb opened the pull request:
Field parsing was getting corrupted b/c no reservation for length of default field value in field header.
Also, shape-type field meta.nullable was hardcoded to false.  Added in the parse of the flag value for shape field.

sainsb committed:
Fixed default value leng on text fieldtype and nullable flag on shape fieldtype

calvinmetcalf commented on lib/fields.js:
```diff
@@ -41,6 +41,11 @@ var dataHeaders = [
     if((out.meta.flag & 1) === 0){
       out.meta.nullable = false;
     }
+    if((out.meta.flag&4)!=0){
```
that should be `!==` not `!=` also spaces around the operators like line 41

calvinmetcalf commented:
looks good though test ?

sainsb commented:
you mean like this https://pbs.twimg.com/media/Cf3I4zQUUAEwBCB.jpg:large ?
OK, added an fgdb with an fc that has a geom field that is nullable and another field that has a default value.  Added a test against said fgdb.

sainsb committed:
Added test and fgdb with fc with null shape field and default field values

sainsb committed:
Added spaces and another equal sign
